k4FWCore/components/UniqueIDGenSvc.h

```diff
@@ -7,6 +7,10 @@
 #include <GaudiKernel/Service.h>
 #include "k4Interface/IUniqueIDGenSvc.h"
 
+
+// Generate a random, unique*, reproducible number using
+// event number, event run, and algorithm name.
+// To be used as ID across events and algorithms.
```

Comment:
```diff
-// Generate a random, unique*, reproducible number using
-// event number, event run, and algorithm name.
-// To be used as ID across events and algorithms.
+// Generate a unique, reproducible number using
+// event number, run number, algorithm name, and global seed
+// To be used as reproducible per event seed for random number generators used inside algorithms.
```
Actually that "global seed" is not yet implemented

Reply:
I.e. the service should have a seed property that allows one to globally change all the specific seeds, but same global  seed results in otherwise same unique seeds.